admin-dev/themes/new-theme/js/pages/product/edit/product-form-model.ts

```diff
@@ -241,4 +253,44 @@ export default class ProductFormModel {
 
     return taxRate.dividedBy(100).plus(1);
   }
+
+  // update tax included price and tax rates help label
+  private updateTaxRulesGroupInfo(taxRatio: BigNumber) {
+    const isTaxEnabled = this.$taxRuleGroupHelpLabel.data('is-tax-enabled');
+
+    if (!isTaxEnabled) {
+      return;
+    }
+    const stateIsoCode = this.getStateIsoCode();
+    const priceTaxExcluded = this.mapper.getBigNumber('price.priceTaxExcluded') ?? new BigNumber(0);
+    const ecotaxTaxIncluded = this.mapper.getBigNumber('price.ecotaxTaxIncluded') ?? new BigNumber(0);
+    this.mapper.set(
+      'price.priceTaxIncluded',
+      priceTaxExcluded.times(taxRatio).plus(ecotaxTaxIncluded).toFixed(this.precision),
+    );
+    const placeHolderWithoutTax = this.$taxRuleGroupHelpLabel.data('place-holder-without-state');
+    const placeHolderWithTax = this.$taxRuleGroupHelpLabel.data('place-holder-with-state');
```

Comment:
```suggestion
    const placeHolderWithoutState = this.$taxRuleGroupHelpLabel.data('place-holder-without-state');
    const placeHolderWithState = this.$taxRuleGroupHelpLabel.data('place-holder-with-state');
```

Reply:
But you could actually only use one const here:

```js
    const taxPlaceholder = this.$taxRuleGroupHelpLabel.data(stateIsoCode ? 'placeholder-with-state' : 'placeholder-without-state');
    this.$taxRuleGroupHelpLabel.html(
      taxPlaceholder
        .replace(
          new RegExp('_TAX_RATE_HELP_PLACEHOLDER_', 'g'),
          taxRatio.minus(1).times(100).toPrecision(),
        )
        .replace(
          new RegExp('_STATE_ISO_CODE_HELP_PLACEHOLDER_', 'g'),
          stateIsoCode,
        )
    );
```

You replace even if the `_STATE_ISO_CODE_HELP_PLACEHOLDER_` is not in the string you don't care it ill simply do nothing, but at least the code is much smaller and a bit easier to read.